src/components/Payment/Payment.js

```diff
@@ -240,43 +240,72 @@ class Payment extends Component {
   };
 
   gernerateLegalNote = () => {
-    const { order, period } = this.props;
+    const { t, order, period } = this.props;
     const discountApplied = order.discount && order.discount.applied;
     const isInTrial = discountApplied && order.discount.type === 'trial';
     const readablePrice = `${currencyFormat[order.currency]}${
       order.priceBreakdown.offerPrice
-    }${period ? `/${period}` : ''}`;
+    }`;
+    const readablePeriod = `${period ? `/${period}` : ''}`;
 
     return (
       <LegalNoteWrapperStyled>
         <LegalTextStyled>
-          <strong>
-            {discountApplied
-              ? 'After any free trial and/or promotional period'
-              : `By clicking 'Complete purchase'`}
-            , you will be charged {readablePrice} or the then-current price plus
-            applicable taxes on a recurring basis.{' '}
-          </strong>
-          {isInTrial &&
-            'If you do not cancel the service during its free trial period, you will be charged. '}
-          Your subscription will automatically continue until you cancel. To
-          cancel, log into{' '}
-          <a
-            href={getData('CLEENG_MY_ACCOUNT_URL')}
-            style={{
-              textDecoration: getData('CLEENG_MY_ACCOUNT_URL')
-                ? 'underline'
-                : 'none'
-            }}
-          >
-            your account
-          </a>{' '}
-          and click &apos;Manage Subscription&apos;.
+          {(() => {
+            if (discountApplied && isInTrial) {
```

Comment:
as the key here is `legal-notes.trial.period-${period}`, should be checked here if `discountApplied`? 
Maybe we should have here:
1. isInTrial,
2. discountApplied,
3. other cases.
Would be more readable